src/main/scala/org/broadinstitute/dsde/workbench/avram/api/AvramServlet.scala

```diff
@@ -1,69 +1,50 @@
 package org.broadinstitute.dsde.workbench.avram.api
 
-import java.util.logging.Logger
+import java.util.logging.{Level, Logger}
 
-import scala.concurrent.duration.Duration
+import io.circe.Encoder
+import io.circe.syntax._
 import javax.servlet.http.{HttpServletRequest, HttpServletResponse}
-
 import org.broadinstitute.dsde.workbench.avram.Avram
 import org.broadinstitute.dsde.workbench.avram.dataaccess.SamUserInfoResponse
 import org.broadinstitute.dsde.workbench.avram.model.AvramException
+import org.broadinstitute.dsde.workbench.avram.util.AvramResult
+import org.broadinstitute.dsde.workbench.avram.util.AvramResult.unsafeRun
 
-import scala.concurrent.Await
-import scala.concurrent.Future
-import scala.util.{Failure, Success}
+import scala.concurrent.ExecutionContext
 
 
 trait AvramServlet {
 
-  implicit val executionContext = scala.concurrent.ExecutionContext.Implicits.global
+  implicit val executionContext: ExecutionContext = scala.concurrent.ExecutionContext.Implicits.global
 
-  val log = Logger.getLogger(getClass.getName)
+  val log: Logger = Logger.getLogger(getClass.getName)
   private val samDao = Avram.samDao
   private val bearerPattern = """(?i)bearer (.*)""".r
 
-  // TODO: Change the structures of registerOnComplete and handleAuthenticationRequest. Not doing it
-  // TODO: for PR #18 because we're going to be switching to using IO with AvramResult and this will
-  // TODO: likely be changed around anyway.
+  def handleAuthenticatedRequest[T](request: HttpServletRequest, response: HttpServletResponse)
+                                   (f: SamUserInfoResponse => AvramResult[T])
+                                   (implicit encoder: Encoder[T]): Unit = {
+    def writeBody(body: T): Unit = {
+      response.setStatus(HttpServletResponse.SC_OK)
+      response.setContentType("application/json")
+      response.getWriter.write(body.asJson.noSpaces)
+    }
 
-  def registerOnComplete[T](future: Future[T], request: HttpServletRequest, response: HttpServletResponse) = {
-    future onComplete {
-      case Success(result) => {
-        response.setStatus(HttpServletResponse.SC_OK)
-      }
-      case Failure(e: AvramException) => {
-        log.severe(e.getMessage)
-        log.severe(e.getStackTrace.toString)
-        response.setStatus(e.status)
-        response.getWriter.write(e.message)
-      }
-      case Failure(e) => {
-        log.severe(e.getMessage)
-        log.severe(e.getStackTrace.toString)
-        response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR)
-        response.getWriter.write(e.getMessage)
-      }
+    def writeError(e: AvramException): Unit = {
+      log.log(Level.SEVERE, "Unhandled error", e)
+      response.setStatus(e.status)
+      response.getWriter.write(e.message)
```

Comment:
Is `e.message` just a String or do we serialize it as json? Either way we should probably set `Content-Type` accordingly here too.

Reply:
Just a string. Is there any reason to wrap it in json? Intent is to only return messages, not stack traces.